lib/gallery_saver.dart

```diff
@@ -65,10 +65,10 @@ class GallerySaver {
     return result;
   }
 
-  static Future<File> _downloadFile(String url) async {
+  static Future<File> _downloadFile(String url, {Map<String, String> headers}) async {
     print(url);
     http.Client _client = new http.Client();
-    var req = await _client.get(Uri.parse(url));
+    var req = headers != null ? await _client.get(Uri.parse(url), headers: headers) : await _client.get(Uri.parse(url));
```

Comment:
Does this break without the explicit null check? [Seems from the implementation that it's not necessary](https://github.com/dart-lang/http/blob/d5c678cd63c3e9c1d779a09acfa95b7e3af84665/lib/src/base_client.dart#L79).

Reply:
Yeah i overreacted and it doesn't break out without the null check i should correct that thax for highlighting it !